spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientAutoConfiguration.java

```diff
@@ -141,7 +142,6 @@ public StreamBuilder streamBuilder(DataFlowOperations dataFlowOperations){
 	}
 
 	@ConditionalOnProperty(prefix = DataFlowPropertyKeys.PREFIX + "client.authentication", name = "client-id")
-	@Configuration
```

Comment:
Just curious if this removal was necessary?   Seems like this should not have been removed.   Thoughts?

Reply:
This is a Spring Framework 6.x (maybe earlier) best practice. Only top-level config needs to be registered - nested configurations are automatically registered. Look for `With nested @Configuration` in the [@Configuration javadocs](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Configuration.html) for more details.

Reply:
But when I read it, it still shows the `@Configuration` added to the inner static class in the sample  and in the doc, it says it means we don't have to add the `@Import` for the static inner class.   
Or is Glenn just being stubborn.   :-)

Reply:
Ahh yeh, that is not the proper doc for this. Let me back up my comment or we can simply remove this. It was put in by OpenRewrite.

Reply:
I was just making sure rewrite didn't take out the wrong `@Configuration`.

Reply:
Updated - added back in.